tests/timelord/test_new_peak.py

```diff
@@ -43,103 +46,344 @@ async def test_timelord_new_peak_basic(
                 await _validate_and_add_block(b1, blocks[-1])
                 await _validate_and_add_block(b2, blocks[-1])
 
-                await timelord_api.new_peak_timelord(timelord_peak_from_block(blocks[-1], b1, bt.constants))
+                await timelord_api.new_peak_timelord(timelord_peak_from_block(b1, blocks[-1]))
                 assert timelord_api.timelord.new_peak.reward_chain_block.height == blocks[-1].height
 
                 blocks_1 = bt.get_consecutive_blocks(2, blocks)
                 await _validate_and_add_block(b1, blocks_1[-2])
                 await _validate_and_add_block(b1, blocks_1[-1])
-                await timelord_api.new_peak_timelord(timelord_peak_from_block(blocks_1[-2], b1, bt.constants))
-                await timelord_api.new_peak_timelord(timelord_peak_from_block(blocks_1[-1], b1, bt.constants))
+                await timelord_api.new_peak_timelord(timelord_peak_from_block(b1, blocks_1[-2]))
+                await timelord_api.new_peak_timelord(timelord_peak_from_block(b1, blocks_1[-1]))
                 assert timelord_api.timelord.new_peak.reward_chain_block.height == blocks_1[-1].height
 
-                # # new unknown peak, weight less then curr peak
-                # blocks_2 = bt.get_consecutive_blocks(1, blocks)
-                # await _validate_and_add_block(b2, blocks_2[-1])
-                # await timelord_api.new_peak_timelord(timelord_peak_from_block(blocks_2[-1], b2, bt.constants))
-                # assert timelord_api.timelord.new_peak.reward_chain_block.height == blocks_1[-1].height
+                # new unknown peak, weight less then curr peak
+                blocks_2 = bt.get_consecutive_blocks(1, blocks)
+                await _validate_and_add_block(b2, blocks_2[-1])
+                await timelord_api.new_peak_timelord(timelord_peak_from_block(b2, blocks_2[-1]))
+                assert timelord_api.timelord.last_state.last_weight == blocks_1[-1].weight
```

Comment:
Since we use `last_state` here, shouldn't we do something similar to the other tests? (after `new_peak_timelord`, make sure `timelord.timelord.new_peak` is None)

Reply:
we are checking last_state at the other tests as well, the weight is not what we want to assert in some of them